lib/classes/class-db.php

```diff
@@ -0,0 +1,886 @@
+      }
+
+      /**
+       * Getters
+       */
+      public function __get($property) {
+        if ( in_array($property, ['files', 'file_sizes', 'file_meta']) ) {
+          return $this->$property;
+        }
+      }
+
+      /**
+       * Creates or updates DB structure
+       */
+      public function create_db() {
+        $version = get_site_option('sm_db_version');
+
+        if ($version === self::DB_VERSION) {
+          return;
+        }
+
+        $charset_collate = $this->wpdb->get_charset_collate();
+
+        $sql = "CREATE TABLE $this->files (
+          `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+          `post_id` bigint(20) unsigned NULL DEFAULT NULL,
+          `bucket` varchar(255) NOT NULL,
+          `name` varchar(255) NOT NULL,
+          `generation` bigint(16) NOT NULL,
+          `cache_control` varchar(255) NULL DEFAULT NULL,
+          `content_type` varchar(255) NULL DEFAULT NULL,
+          `content_disposition` varchar(100) NULL DEFAULT NULL,
+          `file_size` bigint(20) unsigned NULL DEFAULT NULL,
+          `width` int unsigned NULL DEFAULT NULL,
+          `height` int unsigned NULL DEFAULT NULL,
+          `stateless_version` varchar(20) NOT NULL,
+          `storage_class` varchar(50) NULL DEFAULT NULL,
+          `file_link` text NOT NULL,
+          `self_link` text NOT NULL,
+          PRIMARY KEY (`id`),
+          UNIQUE KEY post_id (post_id)
+        ) $charset_collate;
+
+        CREATE TABLE $this->file_sizes (
+          `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+          `post_id` bigint(20) unsigned NULL DEFAULT NULL,
+          `size_name` varchar(255) NOT NULL,
+          `name` varchar(255) NOT NULL,
+          `generation` bigint(16) NOT NULL,
+          `file_size` bigint(20) unsigned NULL DEFAULT NULL,
+          `width` int unsigned NULL DEFAULT NULL,
+          `height` int unsigned NULL DEFAULT NULL,
+          `file_link` text NOT NULL,
+          `self_link` text NOT NULL,
+          PRIMARY KEY (`id`),
+          KEY post_id (post_id),
+          UNIQUE KEY post_id_size (post_id, size_name)
+        ) $charset_collate;
+
+        CREATE TABLE $this->file_meta (
+          `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+          `post_id` bigint(20) unsigned NULL DEFAULT NULL,
+          `meta_key` varchar(255) NOT NULL,
+          `meta_value` longtext NOT NULL,
+          PRIMARY KEY (`id`),
+          KEY post_id (post_id),
+          UNIQUE KEY post_id_key (post_id, meta_key)
+        ) $charset_collate;";
+
+        require_once(ABSPATH . 'wp-admin/includes/upgrade.php');
+        dbDelta($sql);
+
+        update_site_option('sm_db_version', self::DB_VERSION);
+      }
+
+      /**
+       * Remove custom DB table on plugin uninstall
+       */
+      public function clear_db() {
+        $sql = "DROP TABLE IF EXISTS $this->files, $this->file_sizes, $this->file_meta;";
+        $wpdb->query($sql);
```

Comment:
@balexey88 `global $wpdb` or `$this->wpdb` ?

Reply:
Thanks!

`$this->wpdb`